pypdf/_writer.py

```diff
@@ -1808,85 +1808,101 @@ def removeLinks(self) -> None:  # deprecated
+            i = 0
+            while i < len(content.operations):
+                operands, operator = content.operations[i]
+                if operator in jump_operators:
+                    del content.operations[i]
+                elif operator == b"Do":
+                    if (
+                        del_image
+                        and operands[0] in images
+                        or not del_image
+                        and operands[0] in forms
+                    ):
+                        del content.operations[i]
+                    i += 1
+                else:
+                    i += 1
+
+        try:
+            d = cast(dict, cast(DictionaryObject, page["/Resources"])["/XObject"])
+        except KeyError:
+            d = {}
+        for k, v in d.items():
+            o = v.get_object()
+            try:
+                content: Any = None
+                if del_image and o["/Subtype"] == "/Image":
+                    content = NullObject()
+                    images.append(k)
+                if o["/Subtype"] == "/Form":
+                    forms.append(k)
+                    if isinstance(o, ContentStream):
+                        content = o
+                    else:
+                        content = ContentStream(o, self)
+                        content.update(o.items())
+                        for k1 in ["/Length", "/Filter", "/DecodeParms"]:
+                            try:
+                                del content[k1]
+                            except KeyError:
+                                pass
+                    clean(content)
+                if content is not None:
+                    if isinstance(v, IndirectObject):
+                        self._objects[v.idnum - 1] = content
+                    else:
+                        d[k] = self._add_object(content)
+            except (TypeError, KeyError):
+                pass
+        if "/Contents" in page:
+            content = page["/Contents"].get_object()
+            if not isinstance(content, ContentStream):
+                content = ContentStream(content, page)
+            clean(cast(ContentStream, content))
+            if isinstance(page["/Contents"], ArrayObject):
+                for o in cast(ArrayObject, page["/Contents"]):
+                    self._objects[o.idnum - 1] = NullObject()
+            try:
+                self._objects[
+                    cast(IndirectObject, page["/Contents"].indirect_reference).idnum - 1
+                ] = NullObject()
+            except AttributeError:
+                pass
+            page[NameObject("/Contents")] = self._add_object(content)
+
     def remove_images(self, ignore_byte_string_object: bool = False) -> None:
         """
         Remove images from this output.
 
         Args:
-            ignore_byte_string_object: optional parameter
-                to ignore ByteString Objects.
+            ignore_byte_string_object: obsolete
```

Comment:
Why would anybody set `ignore_byte_string_object=True`? What does that mean - what do ByteString objects represent? Could the fact that this PR removes the parameter break peoples workflows?

If we make it obsolete, I suggest to deprecated this parameter as follows:

1. Change the type / default to `Optional[bool] = None`
2. If the value is not `None`, issue a `DeprecationWarning`

```suggestion
            ignore_byte_string_object: deprecated
```

Reply:
I've just seen your comment:

> I have some troubles to understand the parameter

I guess that is the parameter you meant :smile:  Let's discuss the question what it does / if we should remove it in the main issue :-)